website/static/schemas/config.v2.json

```diff
@@ -0,0 +1,206 @@
+        "runNonNativeBuildHooks": {
+            "type": "boolean",
+            "description": "Whether to run build hooks that are defined for an OS other than the host OS.",
+            "default": false
+        },
+        "preBuildHooks": {
+            "$ref": "#/definitions/buildHooks"
+        },
+        "postBuildHooks": {
+            "$ref": "#/definitions/buildHooks"
+        },
+        "author": {
+            "type": "object",
+            "description": "The application author",
+            "properties": {
+                "name": { "type": "string" },
+                "email": {
+                    "type": "string",
+                    "format": "email"
+                }
+            }
+        },
+        "info": {
+            "type": "object",
+            "description": "Data used to populate manifests and version info.",
+            "properties": {
+                "companyName": {
+                    "type": "string",
+                    "description": "The company name. Defaults to the project name."
+                },
+                "productName": {
+                    "type": "string",
+                    "description": "The product name. Defaults to the project name."
+                },
+                "productVersion": {
+                    "type": "string",
+                    "description": "The version of the product",
+                    "default": "1.0.0"
+                },
+                "copyright": {
+                    "type": "string",
+                    "description": "A copyright string for the product",
+                    "default": "Copyright........."
+                },
+                "comments": {
+                    "type": "string",
+                    "description": "A short comment for the app",
+                    "default": "Built using Wails (https://wails.io)"
+                }
+            }
+        },
+        "nsisType": {
+            "type": "string",
+            "default": "multiple",
+            "description": "Type of NSIS Installer for Windows",
+            "oneOf": [
+                {
+                    "const": "multiple",
+                    "description": "One installer per architecture"
+                },
+                {
+                    "const": "single",
+                    "description": "Single universal installer for all architectures being built"
+                }
+            ]
+        },
+        "obfuscated": {
+            "type": "boolean",
+            "default": false,
+            "description": "Whether the binary should be obfuscated. Uses <https://github.com/burrowers/garble>."
+        },
+        "garbleargs": {
+            "type": "boolean",
+            "description": "The arguments to pass to the garble command when using the obfuscated flag"
+        }
+    },
+    "dependencies": {
+        "garbleargs": ["obfuscated"]
+    },
+    "definitions": {
+        "OsHook": {
+            "title": "GOOS/*",
```

Comment:
So these osHook keys are actually the format of the keys rather than explicit values, EG: `windows/*` or `linux/arm7`. I'm not sure how you represent that in JSONSchema!

Reply:
True. I could have validated it against a list of all possible _os_ and _arch_ values. Instead, I chose to add some dummy keys for explanation (lines 182 - 190). The keys are matched against a RegExp at lines 192 - 202.

Just FYI, the `title` key is mainly used for intellisense & documentation. It's doesn't mean you actually use `"title"` in the schema.